api4/user.go

```diff
@@ -3307,3 +3308,26 @@ func getRecentSearches(c *Context, w http.ResponseWriter, r *http.Request) {
 		c.Logger.Warn("Error while writing response", mlog.Err(err))
 	}
 }
+
+func deleteRecentSearch(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireUserId()
+	if c.Err != nil {
+		return
+	}
+
+	if !c.App.SessionHasPermissionToUser(*c.AppContext.Session(), c.Params.UserId) {
+		c.SetPermissionError(model.PermissionEditOtherUsers)
+		return
+	}
+
+	var params model.SearchParams
+	if jsonErr := json.NewDecoder(r.Body).Decode(&params); jsonErr != nil {
+		c.Err = model.NewAppError("deleteRecentSearch", "api.post.search_posts.invalid_body.app_error", nil, "", http.StatusBadRequest).Wrap(jsonErr)
```

Comment:
Could you add a new error message specific to this API? To do that, you'll have to add a new translation string for `api.post.delete_recent_search.invalid_body.app_error` to `i18n/en.json` and then use that ID here